frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -1008,6 +1015,13 @@ export const experimentLogic = kea<experimentLogicType>([
                 },
             }),
             submit: () => {
+                // First step - don't submit, only advance the step
```

Comment:
cc: @Twixes @benjackwhite - not sure if there's some other better way of doing multi-step forms with kea?

Reply:
Ah you're right, this indeed wouldn't work with validation in multiple steps.

The issue with having the Submit button on the last step is that you don't validate until you reach it. There could be errors in any given step, and so we want to show them while still being on that step.

Reply:
Your suggestions are correct Neil, but I don't know if there's an ideal intended workflow for multi-step forms, as that's not a common use case of ours 😄 Kea BDFL @mariusandra might know something though

Reply:
@jurajmajerik good point - that is ok, we can move from errors on submit to errors on touch - this will show validation errors as they happen, and on the next button, we can prevent moving on if there are validation errors on the current page.

```
// whether we show errors after touch (true) or submit (false)
      showErrorsOnTouch: true,
      
      // show errors even without submitting first
      alwaysShowErrors: false,
```

Reply:
I'd say breaking each step/page into its own form is the way to go 🤔

Reply:
ooh interesting. That does complicate things a bit around saving state from submitted forms in a previous step, because our API wouldn't accept an experiment without required params.

I'd for now prefer one form + showing errors on each page, but if that gets too wonky then go multiple forms. (I don't expect it to, you can reuse this helper `hasFormErrors` that I appropriated from kea internals here: https://github.com/PostHog/posthog/pull/20537  - and the trick there to scroll to errors (see surveyEdit.tsx) )

But that's just my preference, your call to make @jurajmajerik

Reply:
One other option is extending https://github.com/keajs/kea-forms/blob/main/src/builder.ts and adding support for `steps` as a first class feature :D.